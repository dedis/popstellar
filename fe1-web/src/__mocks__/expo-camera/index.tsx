import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';

let onBarCodeScanned: Function | null | undefined = null;

export const fireScan = (data: string) => onBarCodeScanned && onBarCodeScanned({ data });

export enum CameraType {
  front = 'front',
  back = 'back',
}

export enum PermissionStatus {
  DENIED = 'denied',
}

export type BarCodeScanningResult = {
  data: string;
};

const Camera = (props: IPropTypes) => {
  ({ onBarCodeScanned } = props);
  return <View />;
};

// To match the PermissionResponse interface
const response = {
  canAskAgain: true,
  expires: 'never',
  granted: true,
  status: 'granted',
};

Camera.useCameraPermissions = () => [response, () => Promise.resolve(response)];
Camera.isAvailableAsync = () => Promise.resolve(true);
Camera.getAvailableCameraTypesAsync = () => Promise.resolve([CameraType.front, CameraType.back]);

const propTypes = {
  onBarCodeScanned: PropTypes.func,
};
Camera.prototype = propTypes;

type IPropTypes = PropTypes.InferProps<typeof propTypes>;
export default Camera;
export { Camera };
