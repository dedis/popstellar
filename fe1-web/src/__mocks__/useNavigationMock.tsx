/**
 * Mocks the React navigation when using useNavigation for tests.
 */
export const mockNavigate = jest.fn();
export const mockAddListener = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...actualNavigation,
    useNavigation: () => ({
      addListener: () => mockAddListener,
      navigate: mockNavigate,
    }),
  };
});
