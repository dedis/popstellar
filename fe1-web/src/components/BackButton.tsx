import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import PropTypes from 'prop-types';
import { Pressable, View } from 'react-native';
import { useNavigation } from '@react-navigation/native';

/**
 * UI button to go back in navigation. You can specify the string corresponding to where it should
 * navigate along with its size in its properties.
 */
const BackButton = (props: IPropTypes) => {
  const { navigationTabName, size, testID } = props;
  const navigation = useNavigation();

  return (
    // The view is there to set the button's clickable layout correctly.
    <View style={{ width: size }}>
      <Pressable onPress={() => navigation.navigate(navigationTabName)} testID={testID}>
        <Ionicons name="chevron-back-outline" size={size} />
      </Pressable>
    </View>
  );
};

const propTypes = {
  navigationTabName: PropTypes.string.isRequired,
  size: PropTypes.number,
  testID: PropTypes.string,
};

type IPropTypes = {
  navigationTabName: string;
  size: number;
  testID: string;
};

BackButton.defaultProps = {
  size: 23,
  testID: 'backButton',
};

BackButton.propTypes = propTypes;

export default BackButton;
