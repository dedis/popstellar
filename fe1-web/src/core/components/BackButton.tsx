import { useNavigation } from '@react-navigation/core';
import { StackNavigationProp } from '@react-navigation/stack';
import React from 'react';

import { AppParamList } from 'core/navigation/typing/AppParamList';
import { Color, Icon } from 'core/styles';

import PoPIcon from './PoPIcon';
import PoPTouchableOpacity from './PoPTouchableOpacity';

type NavigationProps = StackNavigationProp<AppParamList>;

const BackButton = () => {
  const navigation = useNavigation<NavigationProps>();

  return (
    <PoPTouchableOpacity onPress={navigation.goBack}>
      <PoPIcon name="arrowBack" color={Color.inactive} size={Icon.size} />
    </PoPTouchableOpacity>
  );
};

export default BackButton;
