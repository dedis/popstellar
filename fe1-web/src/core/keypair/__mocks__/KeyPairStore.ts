import { jest } from '@jest/globals';

import { mockKeyPair } from '__tests__/utils';

import { KeyPair } from '../../objects';

let keyPair: KeyPair = mockKeyPair;

export namespace KeyPairStore {
  export const store = jest.fn<(kp: KeyPair) => void>((kp: KeyPair) => {
    keyPair = kp;
  });

  export const get = jest.fn<() => KeyPair>(() => keyPair);

  export const getPublicKey = () => get().publicKey;

  export const getPrivateKey = () => get().privateKey;
}
