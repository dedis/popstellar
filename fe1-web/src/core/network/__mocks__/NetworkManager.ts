import { SendingStrategy } from '../strategies/ClientMultipleServerStrategy';

const NetworkManagerModule = jest.requireActual('../NetworkManager');

// expose the correct API
export const { getNetworkManager } = NetworkManagerModule;

// and also one that allows changing the used strategy
const { NetworkManager } = NetworkManagerModule.TEST_ONLY_EXPORTS;
export const getMockNetworkManager = (strategy: SendingStrategy) => new NetworkManager(strategy);
