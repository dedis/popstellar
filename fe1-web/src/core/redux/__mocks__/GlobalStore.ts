import { configureStore } from '@reduxjs/toolkit';
import { combineReducers, Store } from 'redux';
import thunkMiddleware from 'redux-thunk';

import { walletReducer } from 'features/wallet/reducer';

const reducers = combineReducers({
  ...walletReducer,
});

const store: Store = configureStore({
  reducer: reducers,
  middleware: [thunkMiddleware],
});

export const getStore = () => store;
export const dispatch: typeof store.dispatch = (x: any) => store.dispatch(x);
