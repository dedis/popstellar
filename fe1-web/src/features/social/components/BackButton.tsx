import { Ionicons } from '@expo/vector-icons';
import { CompositeScreenProps } from '@react-navigation/core';
import { useNavigation } from '@react-navigation/native';
import { StackScreenProps } from '@react-navigation/stack';
import PropTypes from 'prop-types';
import React from 'react';
import { Pressable, View } from 'react-native';

import { AppParamList } from 'core/navigation/typing/AppParamList';
import { LaoParamList } from 'core/navigation/typing/LaoParamList';
import { SocialParamList } from 'core/navigation/typing/SocialParamList';
import { SocialSearchParamList } from 'core/navigation/typing/SocialSearchParamList';
import STRINGS from 'resources/strings';

type NavigationProps = CompositeScreenProps<
  StackScreenProps<SocialSearchParamList, typeof STRINGS.social_media_navigation_tab_user_profile>,
  CompositeScreenProps<
    StackScreenProps<SocialParamList, typeof STRINGS.social_media_navigation_tab_search>,
    CompositeScreenProps<
      StackScreenProps<LaoParamList, typeof STRINGS.navigation_social_media>,
      StackScreenProps<AppParamList, typeof STRINGS.navigation_app_lao>
    >
  >
>;

/**
 * UI button to go back in navigation. You can specify the string corresponding to where it should
 * navigate along with its size in its properties.
 */
const BackButton = (props: IPropTypes) => {
  const { navigationTabName, size, testID } = props;

  const navigation = useNavigation<NavigationProps['navigation']>();

  return (
    // The view is there to set the button's clickable layout correctly.
    <View style={{ width: size }}>
      <Pressable onPress={() => navigation.navigate(navigationTabName)} testID={testID}>
        <Ionicons name="chevron-back-outline" size={size} />
      </Pressable>
    </View>
  );
};

const propTypes = {
  navigationTabName: PropTypes.string.isRequired,
  size: PropTypes.number,
  testID: PropTypes.string,
};

type IPropTypes = {
  navigationTabName: typeof STRINGS.social_media_navigation_tab_attendee_list;
  size: number;
  testID: string;
};

BackButton.defaultProps = {
  size: 23,
  testID: 'backButton',
};

BackButton.propTypes = propTypes;

export default BackButton;
