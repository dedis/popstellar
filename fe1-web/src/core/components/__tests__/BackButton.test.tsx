import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';

import { mockNavigate } from '__mocks__/useNavigationMock';

import BackButton from '../BackButton';

const mockHomeTab = 'Home';

beforeEach(() => {
  mockNavigate.mockClear();
});

describe('BackButton', () => {
  it('renders correctly', () => {
    const component = render(<BackButton navigationTabName={mockHomeTab} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('navigates correctly', () => {
    const button = render(<BackButton navigationTabName={mockHomeTab} />).getByTestId('backButton');
    fireEvent.press(button);
    expect(mockNavigate).toHaveBeenCalledWith(mockHomeTab);
    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });
});
