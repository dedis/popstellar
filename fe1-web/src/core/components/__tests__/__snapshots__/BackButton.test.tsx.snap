// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders correctly 1`] = `
<View
  style={
    Object {
      "width": 23,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="backButton"
  >
    {"name":"chevron-back-outline","size":23}
  </View>
</View>
`;
